src/gt4py/next/ffront/foast_passes/type_deduction.py

```diff
@@ -717,7 +717,7 @@ def visit_Call(self, node: foast.Call, **kwargs) -> foast.Call:
             isinstance(new_func.type, ts.FunctionType)
             and not type_info.is_concrete(return_type)
             and isinstance(new_func, foast.Name)
-            and new_func.id in fbuiltins.FUN_BUILTIN_NAMES
+            and new_func.id in (fbuiltins.FUN_BUILTIN_NAMES + ["as_offset"])
```

Comment:
lets introduce a EXPERIMENTAL_FUN_BUILTIN_NAMES in ffront.experimental to avoid having this string here.